test/benchmarks/material/radio/radio.perf-spec.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {$, browser} from 'protractor';
+import {runBenchmark} from '@angular/dev-infra-private/benchmark/driver-utilities';
+
+describe('radio button performance benchmarks', () => {
+  beforeAll(() => {
+    browser.rootEl = '#root';
+  });
+
+  it('renders two radio buttons', async() => {
+    await runBenchmark({
+      id: 'render-two-radio-buttons',
+      url: '',
+      ignoreBrowserSynchronization: true,
+      params: [],
+      prepare: async () => await $('#hide-two').click(),
+      work: async () => await $('#show-two').click(),
+    });
+  });
+
+  it('renders ten radio buttons', async() => {
+    await runBenchmark({
+      id: 'render-ten-radio-buttons',
+      url: '',
+      ignoreBrowserSynchronization: true,
+      params: [],
+      prepare: async () => await $('#hide-ten').click(),
+      work: async () => await $('#show-ten').click(),
+    });
+  });
+
+  it('changing between radio buttons', async() => {
+    await runBenchmark({
+      id: 'click-radio-button',
+      url: '',
+      ignoreBrowserSynchronization: true,
+      params: [],
+      setup: async() => await $('#show-two').click(),
+      work: async () => await $('.mat-radio-button:not(.mat-radio-checked)').click(),
```

Comment:
Optional nit: for the sake of stability, it might be better always selecting a given one? e.g. in setup we also always click the first, and then in `work` we always "check" the second?

Reply:
I think you have confused `setup` and `prepare`. Setup gets called just once before the sampling begins. `prepare` gets called before each sample and `work` is the code that gets sampled.

In this test, `setup` is showing two radio buttons (they are hidden by default), and `work` is clicking the radio button that is not selected/checked.

Let me know if I misunderstood your comment or if you have any other followup :+1:

Reply:
Yeah, I meant that we make the benchmark less reliant on the `:not` selector, or the page's state. Instead, we always click one checkbox in `prepare` and the other one (that is then unchecked) in `work` (for the measuring).